### ISSUE TITLE
Broken build on Ubuntu 18.04

### ISSUE THREAD

**Issue by michael-uman** [westes/flex#428]
When I attempt to build Flex on Ubuntu 18.04 I encounter the following build error. Cannot get around this...

```
./stage1flex   -o stage1scan.c ./scan.l
mv -f .deps/flex-dfa.Tpo .deps/flex-dfa.Po
mv -f .deps/flex-main.Tpo .deps/flex-main.Po
Makefile:1696: recipe for target 'stage1scan.c' failed
make[2]: *** [stage1scan.c] Segmentation fault (core dumped)
make[2]: *** Waiting for unfinished jobs....
mv -f .deps/flex-gen.Tpo .deps/flex-gen.Po
make[2]: Leaving directory '/home/developer/development/misc/flex-2.6.4/src'
Makefile:546: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/home/developer/development/misc/flex-2.6.4/src'
Makefile:533: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
```

**Comment by michael-uman** [comment]
A little more information regarding this issue:

Distro : Ubuntu 18.04 LTS
Uname : Linux spaceinvader-dev 5.0.0-29-generic #31~18.04.1-Ubuntu SMP Thu Sep 12 18:29:21 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
vendor_id	: GenuineIntel
cpu family	: 6
model		: 85
model name	: Intel(R) Core(TM) i9-9900X CPU @ 3.50GHz
stepping	: 4
cpu MHz		: 3504.002

flex : 2.6.4

**Comment by michael-uman** [comment]
I dont have time to debug this but my initial impression is that the function `allocate_array` is returning a pointer to memory which has not been properly allocated. 

```
140	/* allocate_array - allocate memory for an integer array of the given size */
141	
142	void   *allocate_array (int size, size_t element_size)
143	{
144		void *mem;
145	#if HAVE_REALLOCARRAY
146		/* reallocarray has built-in overflow detection */
147		mem = reallocarray(NULL, (size_t) size, element_size);
(gdb) l
148	#else
149		size_t num_bytes = (size_t) size * element_size;
150		mem = (size && SIZE_MAX / (size_t) size < element_size) ? NULL :
151			malloc(num_bytes);
152	#endif
153		if (!mem)
154			flexfatal (_
155				   ("memory allocation failed in allocate_array()"));
156	
157		return mem;
(gdb) q
```
It appears `HAVE_REALLOCARRAY` is defined and it is returning a pointer to unallocated memory.

**Comment by Explorer09** [comment]
Thank you, but this is a known issue of flex-2.6.4.
Duplicate of issue #241 (already fixed in master) and PR #262.
**Workaround:** `./configure CFLAGS='-g -O2 -D_GNU_SOURCE'`
